src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -252,14 +252,51 @@ private void OnAgentExit(Process process, Guid agentId)
                 case AgentExitCodes.UNABLE_TO_LOCATE_AGENCY:
                     errorMsg = "Remote test agent unable to locate agency process.";
                     break;
+                //case AgentExitCodes.STACK_OVERFLOW_EXCEPTION:
+                //    errorMsg = "Remote test agent was terminated due to a StackOverFlow exception in one of the tests.";
+                //    break;
                 default:
-                    errorMsg = $"Remote test agent exited with non-zero exit code {process.ExitCode}";
+                    //errorMsg = $"Remote test agent exited with non-zero exit code {process.ExitCode}";
+                    errorMsg = GetDefaultError(process.ExitCode);
```

Comment:
This is just a draft and an afterthought. My intention is to create something, given the environment and exit code would give me back the most likely error message.